nifi-docs/src/main/asciidoc/developer-guide.adoc

```diff
@@ -2637,19 +2637,83 @@ logged to help avoid this bad practice.
 
 
 [[deprecation]]
-== Deprecating a Component
+== Deprecating Components and Features
+Deprecating features is an important part of the software development lifecycle, providing an upgrade path for
+developers and users to follow.
+
+Apache NiFi follows the link:https://semver.org[Semantic Versioning Specification 2.0.0^] for features identified as
+part of the public version contract according to
+link:https://cwiki.apache.org/confluence/display/NIFI/Version+Scheme+and+API+Compatibility[Version Scheme]
+documentation.
+
+Components and features that fit under the public version contract require deprecation marking prior to removal. This
+approach allows developers to implement changes as part of minor version upgrades, in preparation for future removal
+of features in a subsequent major version.
```

Comment:
In the larger context of this section, this references to removing features that would cause breaking changes. That is the reason for the opening sentence referencing "features that fit under the public version contract." There may be some theoretical edge case where a feature needs to be removed for some other reason, like a security vulnerability, but this sentence states the standard approach.